android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DevCycleClient.kt

```diff
@@ -158,15 +159,8 @@ class DevCycleClient private constructor(
             if (type == "refetchConfig" || type == "") { // Refetch the config if theres no type
                 refetchConfig(true, lastModified, etag)
             }
-        }), URI(config?.sse?.url)).executor(this.getValidExecutorService()).build()
-        eventSource?.start()
-    }
-
-    private fun getValidExecutorService(): ExecutorService? {
-        if (executorService == null || executorService?.isTerminated == true) {
-            executorService = Executors.newSingleThreadExecutor()
-        }
-        return executorService
+        }), EventSource.Builder(URI(config?.sse?.url))).build()
+        backgroundEventSource?.start()
```

Comment:
Does this block?

Reply:
I don't believe so, as its using a background thread.

Reply:
looking at the lib - looks like yes